### PR TITLE
add vds resource

### PR DESCRIPTION
**Pull request by klaricch**

### CODE REVIEW

File: gnomad/resources/resource_utils.py [hunk 1]
```diff
@@ -159,6 +159,38 @@ def import_resource(self, overwrite: bool = True, **kwargs) -> None:
...
+    :param path: The VariantDataset path (typically ending in .vds)
+    :param import_args: Any sources that are required for the import and need to be kept track of and/or passed to the import_func (e.g. .vcf path for an imported VCF)
+    :param import_func: A function used to import the VariantDataset. `import_func` will be passed the `import_args` dictionary as kwargs.
+    """
+
+    expected_file_extensions: List[str] = [".vds"]
+
+    def vds(self, force_import: bool = False) -> hl.vds.VariantDataset:
+        """
+        Read and return the Hail VariantDataset resource.
```
Comment (jkgoodrich) [inline gnomad/resources/resource_utils.py:174]:
```suggestion
        Read and return the Hail VariantDataset resource.
        
```

File: gnomad/resources/resource_utils.py [hunk 2]
```diff
@@ -159,6 +159,38 @@ def import_resource(self, overwrite: bool = True, **kwargs) -> None:
...
+        :return: Hail VariantDataset resource
+        """
+        if self.path is None or force_import:
+            return self.import_func(**self.import_args)
+        else:
+            return hl.read_vds(self.path)
```
Comment (jkgoodrich) [inline gnomad/resources/resource_utils.py:180]: I think this is actually:
```suggestion
            return hl.vds.read_vds(self.path)
```

File: gnomad/resources/resource_utils.py [hunk 3]
```diff
@@ -159,6 +159,38 @@ def import_resource(self, overwrite: bool = True, **kwargs) -> None:
...
+
+    def import_resource(self, overwrite: bool = True, **kwargs) -> None:
+        """
+        Import the VariantDataset resource using its import_func and writes it in its path.
```
Comment (jkgoodrich) [inline gnomad/resources/resource_utils.py:184]:
```suggestion
        Import the VariantDataset resource using its import_func and writes it in its path.
        
        
```

File: gnomad/resources/resource_utils.py [hunk 4]
```diff
@@ -339,6 +371,21 @@ def __init__(self, default_version: str, versions: Dict[str, MatrixTableResource
         super().__init__(default_version, versions)
 
 
+class VersionedVariantDatasetResource(BaseVersionedResource):
+    """
+    Versioned VariantDataset resource.
```
Comment (jkgoodrich) [inline gnomad/resources/resource_utils.py:376]:
```suggestion
    Versioned VariantDataset resource.
    
```

File: gnomad/resources/resource_utils.py [hunk 5]
```diff
@@ -339,6 +371,21 @@ def __init__(self, default_version: str, versions: Dict[str, MatrixTableResource
...
+    The attributes (path, import_args and import_func) of the versioned resource are those of the default version of the resource.
+    In addition, all versions of the resource are stored in the `versions` attribute.
+    :param default_version: The default version of this VariantDataset resource (must to be in the `versions` dict)
```
Comment (jkgoodrich) [inline gnomad/resources/resource_utils.py:379]:
```suggestion
    :param default_version: The default version of this VariantDataset resource (must to be in the `versions` dict)
    
```

**Review by jkgoodrich — CHANGES_REQUESTED** [review]
A few small things

**Comment by jkgoodrich** [comment]
Also, I think you need to run Black

**Review by jkgoodrich — APPROVED** [review]
LGTM

**Comment by nawatts** [comment]
The [Pylint failure](https://github.com/broadinstitute/gnomad_methods/runs/4357803799?check_suite_focus=true) is resolved in #425.

**Pull request merged**